### PR TITLE
python310Packages.torch-bin: correct meta.platforms

### PR DESCRIPTION
**Pull request by dotlambda**
###### Description of changes
We no longer need to rely on throw. See https://github.com/NixOS/nixpkgs/pull/215721#issuecomment-1426856654

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by mweinelt — APPROVED** [review]

**Comment by dotlambda** [comment]
Hmm, it seems like we have do add `or null` or something but then we get
```
$ NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM=1 nix-build -A python310.pkgs.torch-bin --system aarch64-linux
error: value is null while a set was expected

       at pkgs/development/python-modules/torch/bin.nix:33:9:

           32|
           33|   src = fetchurl srcs."${stdenv.system}-${pyVerNoDot}" or null;
             |         ^
           34|
```
So maybe just leave the `throw`?

**Comment by mweinelt** [comment]
Yep, apparently having both would yield the ideal outcome.

File: pkgs/development/python-modules/torchvision/bin.nix [hunk 1]
```diff
@@ -61,7 +61,7 @@ in buildPythonPackage {
     # https://www.intel.com/content/www/us/en/developer/articles/license/onemkl-license-faq.html
     license = licenses.bsd3;
     sourceProvenance = with sourceTypes; [ binaryNativeCode ];
-    platforms = platforms.linux;
+    platforms = [ "x86_64-linux" ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/torchvision/bin.nix:64]:
```suggestion
    platforms = [ "aarch64-darwin" "x86_64-darwin" "x86_64-linux" ];
```

File: pkgs/development/python-modules/torchaudio/bin.nix [hunk 2]
```diff
@@ -51,7 +51,7 @@ buildPythonPackage rec {
     # https://www.intel.com/content/www/us/en/developer/articles/license/onemkl-license-faq.html
     license = licenses.bsd3;
     sourceProvenance = with sourceTypes; [ binaryNativeCode ];
-    platforms = platforms.linux;
+    platforms = [ "x86_64-linux" ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/torchaudio/bin.nix:54]:
```suggestion
    platforms = [ "aarch64-darwin" "x86_64-darwin"  "x86_64-linux" ];
```

File: pkgs/development/python-modules/torchvision/bin.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/development/python-modules/torchvision/bin.nix:64]: patchelf doesn't work on Darwin, so the package as is doesn't support Darwin

Comment (mweinelt) [reply pkgs/development/python-modules/torchvision/bin.nix:64]: Ah, I went from the list of hashes in `binary-hashes.nix`.

**Pull request merged**